Add deduction guides for blocked_nd_range (#651)
oneapi-src/oneAPI-spec: source/elements/oneTBB/source/algorithms/blocked_ranges/blocked_nd_range_cls.rst

                // Access
                bool is_divisible() const;
                const dim_range_type& dim(unsigned int dimension) const;
            }; // class blocked_nd_range

            // Deduction Guides
            template <typename Value, typename... Values>
            blocked_nd_range(blocked_range<Value>, blocked_range<Values>...)
            -> blocked_nd_range<Value, 1 + sizeof...(Values)>;

            template <typename Value, unsigned int N>
            blocked_nd_range(const Value (&)[N], typename blocked_nd_range<Value, N>::size_type = 1)
            -> blocked_nd_range<Value, N>;

>             template </*see-below*/>

Reviewer: I prefer keeping the previous: 

`template <typename Value>`
`blocked_nd_range(/*a set of braced-init-lists of Value objects*/)`
`        -> blocked_nd_range<Value, /*number of braced-init-lists*/>;`

And then the detailed description provides the two options for how to do that. What's currently here seems even less clear.

Reviewer: You might add the "see-below" comment in addition, like `blocked_nd_range(/*a set of braced-init-lists of Value objects (see-below)*/)`

Author: Applied the combined approach with the comment below.

**Returns:** ``blocked_range`` containing the value space along the dimension specified by the argument.

Deduction Guides
----------------

.. code:: cpp

    template <typename Value, typename... Values>
    blocked_nd_range(blocked_range<Value>, blocked_range<Values>...)
    -> blocked_nd_range<Value, 1 + sizeof...(Values)>;

**Effects:**: Enables deduction when a set of ``blocked_range`` objects is passed to the ``blocked_nd_range`` constructor.

**Constraints:**: Participates in overload resolution only if all of the types in ``Values`` are same as ``Value``.

.. code:: cpp

    template <typename Value>
    blocked_nd_range(/*a set of braced-init-lists of Value objects*/)
    -> blocked_nd_range<Value, /*number of braced-init-lists*/>;

**Effects:**: Enables deduction when a set of ``blocked_range`` objects is provided as braced-init-lists to the ``blocked_nd_range`` constructor.

The exact signature of this deduction guide is unspecified.

**Constraints:**: Participates in overload resolution only if the number of braced-init-lists provided is more than ``1`` and each list contains

> ``2`` or ``3`` elements of the same type, corresponding to ``blocked_range`` constructors with 2 and 3 arguments, respectively.

Reviewer 2: ```suggestion
``2`` or ``3`` elements of the same type.
```

Is there anything important in the part that I suggest to remove?

> **Example:**: ``blocked_nd_range range({0, 100}, {0, 100, 50})`` should deduce ``range`` as ``blocked_nd_range<int, 3>``.

Reviewer 2: ```suggestion
**Example:**: ``blocked_nd_range range({0, 100}, {0, 100, 50})`` should deduce ``range`` as ``blocked_nd_range<int, 2>``.
```
3 is definitely wrong for the given example

**Returns:** ``blocked_range`` containing the value space along the dimension specified by the argument.

Deduction Guides
----------------

.. code:: cpp

    template <typename Value, typename... Values>
    blocked_nd_range(blocked_range<Value>, blocked_range<Values>...)
    -> blocked_nd_range<Value, 1 + sizeof...(Values)>;

**Effects:**: Enables deduction when a set of ``blocked_range`` objects is passed to the ``blocked_nd_range`` constructor.

**Constraints:**: Participates in overload resolution only if all of the types in ``Values`` are same as ``Value``.

.. code:: cpp

    template <typename Value, unsigned int N>
    blocked_nd_range(const Value (&)[N], typename blocked_nd_range<Value, N>::size_type = 1)
    -> blocked_nd_range<Value, N>;

**Effects:**: Enables deduction from a single C array object indicating a set of dimension sizes.

.. code:: cpp

    template <typename Value, /*a template parameter pack*/... Xs>
    blocked_nd_range(/*a type deducible from a braced-enclosed initializer list of Value objects*/... args)
    -> blocked_nd_range<Value, sizeof...(args)>;

> **Effects:**: Enables deduction when a set of ``blocked_range`` objects is provided as brace-enclosed initializer lists to the ``blocked_nd_range`` constructor.

Reviewer 3: ```suggestion
**Effects:**: Enables deduction when a set of ``blocked_range`` objects is provided as braced initialization lists to the ``blocked_nd_range`` constructor.
```
